managers/backup-manager/BackupStatusPoller.js

```diff
@@ -0,0 +1,277 @@
+          const currentTime = new Date();
+          const backupTriggeredDuration = (currentTime - new Date(opts.started_at)) / 1000;
+          return Promise
+            .try(() => eventmesh
+              .apiServerClient
+              .patchResource({
+                resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                resourceId: opts.backup_guid,
+                status: {
+                  response: operationStatusResponse
+                }
+              }))
+            .then(() => {
+              const lockDeploymentMaxDuration = eventmesh.lockManager.getLockTTL(operationName);
+              if (backupTriggeredDuration > lockDeploymentMaxDuration) {
+                //Operation timed out
+                if (!opts.abortStartTime) {
+                  //Operation not aborted. Aborting operation and with abort start time
+                  let abortStartTime = new Date().toISOString();
+                  return eventmesh.apiServerClient.patchResource({
+                      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                      resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                      resourceId: opts.backup_guid,
+                      status: {
+                        state: CONST.APISERVER.RESOURCE_STATE.ABORT,
+                        response: {
+                          abortStartTime: abortStartTime
+                        }
+                      }
+                    })
+                    .then(() => {
+                      operationStatusResponse.state = CONST.OPERATION.ABORTING;
+                      return operationStatusResponse;
+                    });
+                } else {
+                  // Operation aborted
+                  const currentTime = new Date();
+                  const abortDuration = (currentTime - new Date(opts.abortStartTime));
+                  if (abortDuration < config.backup.abort_time_out) {
+                    logger.info(`${operationName} abort is still in progress on : ${deployment} for guid : ${backup_guid}`);
+                    operationStatusResponse.state = CONST.OPERATION.ABORTING;
+                  } else {
+                    operationStatusResponse.state = CONST.OPERATION.ABORTED;
+                    logger.info(`Abort ${operationName} timed out on : ${deployment} for guid : ${backup_guid}. Flagging ${operationName} operation as complete`);
+                  }
+                  return operationStatusResponse;
+                }
+              } else {
+                // Backup not timedout and still in-porogress
+                return operationStatusResponse;
+              }
+            });
+        }
+      })
+      .then(operationStatusResponse => utils.isServiceFabrikOperationFinished(operationStatusResponse.state) ?
```

Comment:
One more thing @amitmalav does this also ensure that if there is any random error like not able to reach agent or so, it should still continue polling. Only a finished state response from agent or timeout should stop polling. Please confirm it once.

Reply:
Yes, if any other error comes, we just catch and log the error, we don't clear interval.